### ISSUE TITLE
BrierScore possible confusion score/loss

### ISSUE THREAD

**Issue by tlienart** [alan-turing-institute/MLJBase.jl#146]
1. ~~the definition does not seem to match [the wk article](https://en.wikipedia.org/wiki/Brier_score)~~
1. wikipedia effectively  has  it as  a cost function (loss) where  we use it as a score. I'm happy with using it as a score as  it corresponds with the  name... 

Note however that

1.  in R it's implemented  as  a  loss https://www.rdocumentation.org/packages/DescTools/versions/0.99.31/topics/BrierScore
1. same in Sklearn https://scikit-learn.org/stable/modules/generated/sklearn.metrics.brier_score_loss.html
1. see also this  article  https://www.jclinepi.com/article/S0895-4356(09)00363-1/pdf

We might confuse people here if we stick with the score orientation?

**Edit**: on another note, there's a stray factor 2 ? 

```julia
julia> y2 = categorical(["spam", "ham", "ham", "spam", "ham", "ham"]);
julia> L2 = classes(y2[1]);
julia> yhat2 = [UnivariateFinite(L2, v) for v in (
    [.1, .9], [.9, .1], [.8, .2], [.35, .65], [0.2, 0.8], [0.3,0.7]
    )];
julia> mean(BrierScore()(yhat2, y2))
-0.437500
```

sklearn

```python
>>> from sklearn.metrics import brier_score_loss
>>> brier_score_loss([1, 0, 0, 1, 0, 0], [.9, .1, .2, .65, 0.8, 0.7])
0.21875
>>> 0.4375 / 2
0.21875
```

**Comment by ablaom** [comment]
I think the confusion is inevitable.

In the proper scoring rules literature I have looked at, the BrierScore (aka quadratic score) really is a score, ie bigger is better. So I guess it depends on the community. 

The wikipedia entry reflects Brier's original paper (1950) in that it defines what most of us would now call a *loss* but names it a *score*.

Regarding the factor of two: As the wikipedia article points out: *For binary forecasts the original formulation of Brier's "probability score" has twice the value of the score currently known as the Brier score.*  In other words, in the binary case, people have slipped into using a "simplified" formula, which is actually an unnatural choice when you want to define it for the general multiclass case.

Do we want to treat binary as different from every other case of Finite{n}, n>2? I don't see why we should start doing so now.

Note also that there exist generalizations of the BrierScore to distributions other than finite ones. In that case the most natural way to write the formula is a little different but consistent with our implementation in the finite case. See eg, equation (18) in https://www.stat.washington.edu/raftery/Research/PDF/Gneiting2007jasa.pdf . 

I propose we keep the status quo but add a warning to the docstring (and docstring trait) that other conventions (regarding sign and factor of 2) exist.

@tlienart What do you think?

**Comment by tlienart** [comment]
> Note also that there exist generalizations of the BrierScore to distributions other than finite ones. In that case the most natural way to write the formula is a little different but consistent with our implementation in the finite case. See eg, equation (18) in https://www.stat.washington.edu/raftery/Research/PDF/Gneiting2007jasa.pdf .

I think this is important so I'm happy with sticking to  the status quo. 

Seems like the remaining todo is to add a clear message (docstring(s)). I'll  do that.
